starport/services/network/queries.go

```diff
@@ -24,21 +29,50 @@ func (n Network) ChainLaunch(ctx context.Context, id uint64) (networktypes.Chain
 	return networktypes.ToChainLaunch(res.Chain), nil
 }
 
-// ChainLaunches fetches the chain launches from Starport Network
-func (n Network) ChainLaunches(ctx context.Context) ([]networktypes.ChainLaunch, error) {
-	var chainLaunches []networktypes.ChainLaunch
+// ChainLaunchesWithReward fetches the chain launches with rewards from Starport Network
+func (n Network) ChainLaunchesWithReward(ctx context.Context) ([]networktypes.ChainLaunch, error) {
+	g, ctx := errgroup.WithContext(ctx)
+	chainLaunchesChan := make(chan networktypes.ChainLaunch)
 
 	n.ev.Send(events.New(events.StatusOngoing, "Fetching chains information"))
 	res, err := launchtypes.NewQueryClient(n.cosmos.Context).ChainAll(ctx, &launchtypes.QueryAllChainRequest{})
 	if err != nil {
-		return chainLaunches, err
+		return nil, err
 	}
 
-	// Parse fetched chains
+	n.ev.Send(events.New(events.StatusOngoing, "Fetching rewards"))
+	var chainLaunches []networktypes.ChainLaunch
```

Comment:
I don't think we need to use any Go channel inside `ChainLaunchesWithReward`.

We can protect `chainLaunches` with a simple mutex. 

Channels are complex and it's easy to make mistakes with them, we can avoid using channels and raw goroutines as much as it's possible.